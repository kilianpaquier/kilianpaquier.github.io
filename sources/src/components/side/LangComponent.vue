<template>
  <v-container fluid>
    <v-list rounded class="pa-0 text-center text-sm-right">
      <v-list-item-group v-model="selectedItem" mandatory>
        <!-- One item per lang -->
        <v-list-item
          v-for="(lang, i) in langs"
          :key="i"
          @click="changeLang(lang)"
        >
          <!-- Lang title and score -->
          <v-list-item-content class="d-none d-sm-block">
            <v-list-item-title
              v-text="$vuetify.lang.t(`$vuetify.${lang.value}`)"
            />
            <v-list-item-subtitle
              v-text="$vuetify.lang.t(`$vuetify.${lang.info}`)"
            />
          </v-list-item-content>

          <!-- Lang flag -->
          <v-list-item-avatar class="mx-auto ms-sm-4">
            <v-img
              :src="`/img/flags/1x1/${lang.img}`"
              contain
              eager
            />
          </v-list-item-avatar>
        </v-list-item>
      </v-list-item-group>
    </v-list>
  </v-container>
</template>

<script lang="ts">
import Vue from "vue";
import { Lang } from "@/model/entities";
import Component from "vue-class-component";

@Component
export default class LangComponent extends Vue {
  langs: Lang[] = [
      {
          img: "fr.svg",
          value: "langs.fr",
          lang: "fr",
          info: "langs.natal",
      },
      {
          img: "gb.svg",
          value: "langs.en",
          lang: "en",
          info: "langs.professional",
      },
  ] as Lang[];

  get selectedItem() {
      return this.$vuetify.lang.current === "fr" ? 0 : 1;
  }

  set selectedItem(item: number) {
      // Do nothing
  }

  changeLang(item: Lang) {
      this.$vuetify.lang.current = item.lang;
  }
}
</script>
