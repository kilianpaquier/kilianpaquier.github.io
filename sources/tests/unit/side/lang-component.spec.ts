import { createLocalVue, mount, Wrapper } from "@vue/test-utils";
import LangComponent from "@/components/side/LangComponent.vue";
import Vuetify from "vuetify";
import Vue from "vue";

Vue.use(Vuetify);
Vue.config.silent = true;
Vuetify.config.silent = true;

describe("LangComponent.vue", () => {
    const localVue = createLocalVue();

    let vuetify: Vuetify;

    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    let wrapper: Wrapper<any>;

    beforeEach(() => {
        vuetify = new Vuetify();

        wrapper = mount(LangComponent, {
            localVue,
            vuetify,
        });
    });

    it("should create LangComponent", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("should change current lang to fr", () => {
        wrapper.vm.changeLang({
            lang: "fr",
            img: "fr.png",
            info: "test",
            value: "fr",
        });
        expect(wrapper.vm.selectedItem).toEqual(0);
    });

    it("should change current lang to en", () => {
        wrapper.vm.changeLang({
            lang: "en",
            img: "gb.png",
            info: "test",
            value: "en",
        });
        expect(wrapper.vm.selectedItem).toEqual(1);
    });
});
