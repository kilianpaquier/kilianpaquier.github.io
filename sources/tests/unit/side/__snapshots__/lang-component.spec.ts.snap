// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LangComponent.vue should create LangComponent 1`] = `
<div class="container container--fluid">
  <div role="list" class="v-list pa-0 text-center text-sm-right v-sheet theme--light rounded v-list--rounded">
    <div role="listbox" class="v-item-group theme--light v-list-item-group">
      <div tabindex="0" role="option" aria-selected="false" class="v-list-item v-list-item--link theme--light">
        <div class="v-list-item__content d-none d-sm-block">
          <div class="v-list-item__title">$vuetify.langs.fr</div>
          <div class="v-list-item__subtitle">$vuetify.langs.natal</div>
        </div>
        <div class="v-avatar v-list-item__avatar mx-auto ms-sm-4" style="height: 40px; min-width: 40px; width: 40px;">
          <div class="v-image v-responsive theme--light">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--contain" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
      </div>
      <div tabindex="0" role="option" aria-selected="true" class="v-list-item v-item--active v-list-item--active v-list-item--link theme--light">
        <div class="v-list-item__content d-none d-sm-block">
          <div class="v-list-item__title">$vuetify.langs.en</div>
          <div class="v-list-item__subtitle">$vuetify.langs.professional</div>
        </div>
        <div class="v-avatar v-list-item__avatar mx-auto ms-sm-4" style="height: 40px; min-width: 40px; width: 40px;">
          <div class="v-image v-responsive theme--light">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--contain" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
